### PR TITLE
JR2-518 Save jobs inside of transaction template

### PR DESCRIPTION
**Pull request by varusakov**

### CODE REVIEW

**Comment by michael-wirth** [comment]
see also https://www.programcreek.com/java-api-examples/index.php?api=org.springframework.transaction.support.TransactionSynchronizationAdapter

File: src/main/java/ch/admin/seco/service/job/service/impl/JobServiceImpl.java [hunk 1]
```diff
@@ -160,20 +153,11 @@ public CountDTO count(JobSearchRequestDTO jobSearchRequest) {
     }
 
     @Override
-    @Transactional
```
Comment (michael-wirth) [inline src/main/java/ch/admin/seco/service/job/service/impl/JobServiceImpl.java]: Why was the @Transactional Annotation removed? The `TransactionalEventListener` should run inside an open transaction so the event listener know when to run the "AFTER_COMMIT". 

It shouldn't be required to use the flag `(fallbackExecution = true)`

File: src/main/java/ch/admin/seco/service/job/service/event/listener/JobSavedEventListener.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    private final JobMapper jobIndexMapper;
+
+    public JobSavedEventListener(JobSearchRepository jobSearchRepository, JobMapper jobIndexMapper) {
+        this.jobSearchRepository = jobSearchRepository;
+        this.jobIndexMapper = jobIndexMapper;
+    }
+
+    @TransactionalEventListener(fallbackExecution = true)
+    public void onJobSave(JobSavedEvent jobSavedEvent) {
```
Comment (michael-wirth) [inline src/main/java/ch/admin/seco/service/job/service/event/listener/JobSavedEventListener.java]: nice solution. I wasn't aware of that feature!

Comment (varusakov) [reply src/main/java/ch/admin/seco/service/job/service/event/listener/JobSavedEventListener.java]: Thanks!

**Pull request merged**